python/ray/_private/job_manager/tests/test_job_manager.py

```diff
@@ -23,69 +29,78 @@ def check_job_succeeded(job_manager, job_id):
+        wait_for_condition(
+            check_job_succeeded, job_manager=job_manager, job_id=job_id)
+        assert job_manager.get_job_stderr(job_id) == b"error"
 
+    def test_submit_ls_grep(self, job_manager):
+        job_id = job_manager.submit_job("ls | grep test_job_manager.py")
 
-def test_subprocess_exception(job_manager):
-    """
-    Run a python script with exception, ensure:
-    1) Job status is marked as failed
-    2) Job manager can surface exception message back to stderr api
-    3) Job no hanging job supervisor actor
-    4) Empty stdout
-    """
-    job_id = job_manager.submit_job(
-        "python subprocess_driver_scripts/script_with_exception.py")
+        wait_for_condition(
+            check_job_succeeded, job_manager=job_manager, job_id=job_id)
+        assert job_manager.get_job_stdout(job_id) == b"test_job_manager.py"
 
-    wait_for_condition(
-        check_job_failed, job_manager=job_manager, job_id=job_id)
-    stderr = job_manager.get_job_stderr(job_id).decode("utf-8")
-    last_line = stderr.strip().splitlines()[-1]
-    assert last_line == "Exception: Script failed with exception !"
-    assert job_manager._get_actor_for_job(job_id) is None
-    assert job_manager.get_job_stdout(job_id) == b""
+    def test_subprocess_exception(self, job_manager):
+        """
+        Run a python script with exception, ensure:
+        1) Job status is marked as failed
+        2) Job manager can surface exception message back to stderr api
+        3) Job no hanging job supervisor actor
+        4) Empty stdout
+        """
+        job_id = job_manager.submit_job(
+            "python subprocess_driver_scripts/script_with_exception.py")
 
+        wait_for_condition(
+            check_job_failed, job_manager=job_manager, job_id=job_id)
+        stderr = job_manager.get_job_stderr(job_id).decode("utf-8")
+        last_line = stderr.strip().splitlines()[-1]
+        assert last_line == "Exception: Script failed with exception !"
+        assert job_manager._get_actor_for_job(job_id) is None
+        assert job_manager.get_job_stdout(job_id) == b""
 
-def test_submit_with_s3_runtime_env(job_manager):
-    job_id = job_manager.submit_job(
-        "python script.py",
-        runtime_env={"working_dir": "s3://runtime-env-test/script.zip"})
+    def test_submit_with_s3_runtime_env(self, job_manager):
+        job_id = job_manager.submit_job(
+            "python script.py",
+            runtime_env={"working_dir": "s3://runtime-env-test/script.zip"})
 
-    wait_for_condition(
-        check_job_succeeded, job_manager=job_manager, job_id=job_id)
-    assert job_manager.get_job_stdout(
-        job_id) == b"Executing main() from script.py !!"
+        wait_for_condition(
+            check_job_succeeded, job_manager=job_manager, job_id=job_id)
+        assert job_manager.get_job_stdout(
+            job_id) == b"Executing main() from script.py !!"
 
 
 class TestRuntimeEnv:
```

Comment:
Btw, doesn't need to be in this PR, but realized we don't have any tests for running the driver in a pip/conda env -- we should add that to make sure it works (no reason it shouldn't)